File: 14-longest-common-prefix/longest-common-prefix.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function(s) {
    let x=0;
    while(x<s[0].length){
        let ch=s[0][x];
        for(let i=0;i<s.length;i++){
            if(ch!==s[i][x] || x===s[i].length){
                return s[0].substring(0,x);
            }
        }
        ++x;
    }
    return s[0];
};

File: 14-longest-common-prefix/longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        int x=0;
        while(x<strs[0].length()){
            char ch=strs[0].charAt(x);
            for(int i=1;i<strs.length;i++){
                if(x==strs[i].length() || ch!=strs[i].charAt(x)){
                    return strs[0].substring(0,x);
                }
            }
            x++;
        }
        return strs[0];
        
    }
}